cogs/rcon.py

```diff
@@ -0,0 +1,262 @@
+#un implemented.
+
+import discord
+from discord.ext.commands import Cog, command, has_any_role
+from utils import load_config
+import a2s
+import re
+from rcon import Client
+#https://pypi.org/project/rcon/
+
+
+
+global_config = load_config()
+config = global_config
+
+class rcon(Cog):
+
+    #test command
+    @command()
+    async def test(self, ctx):
+        await ctx.trigger_typing()
+        await ctx.send("test")
+    
+    #list all servers
+    @command()
+    @has_any_role('Staff', 'Server Mods', 'Senior Staff', 'Fub')
+    async def servers(self, ctx):
+        allservers = config.servers
+
+        embed=discord.Embed()
+        embed.add_field(name="Availible Servers", value="eu, us, eumvm, usmvm")
+        embed.set_footer(text="TF2M RCON v1")
+
+        await ctx.send(embed=embed)
+
+    #rcon TODO
+    #sm_say works but returns an error message in vsc because there is no return message to discord.
+    @command()
+    @has_any_role('Staff', 'Server Mods', 'Senior Staff', 'Fub')
+    async def rcon(self, ctx, server, *commands):
+
+        #what servers can we rcon
+        servers = ["eu", "us", "eumvm", "usmvm"]
+
+        #only accept valid servers
+        if server in servers:
+
+            #rcon to specific server
+            #eu
+            if server == "eu":
+                command = " ".join(commands)
+                response = await run_rcon(server, command)
+
+                #for status
+                if command == "status":
+                    status_response = await get_status(response)
+                    await ctx.send(status_response)
+                else:
+                #for everything else
+                    if response == "":
+                        await ctx.send("Command sent.")
+                    else:
+                        await ctx.send("```\n" + response + "\n```")
+            #us
+            elif server == "us":
+                command = " ".join(commands)
+                response = await run_rcon(server, command)
+
+                #for status
+                if command == "status":
+
+                    status_response = await get_status(response)
+                    await ctx.send(status_response)
+
+                else:
+                #for everything else
+                    if response == "":
+                        await ctx.send("Command sent.")
+                    else:
+                        await ctx.send("```\n" + response + "\n```")
+            #eumvm
+            elif server == "eumvm":
+                command = " ".join(commands)
+                response = await run_rcon(server, command)
+
+                #for status
+                if command == "status":
+                    await ctx.send("```\n" + response.split('#', 1)[1] + "\n```")
```

Comment:
why is this status handler different than the ones oyu have above. we should have this all be standaized and abstracted into a function

Reply:
this is cause the output of !rcon <server> status on a full server exceeds discords limit for characters. It's literally only status that has this issue so that's why I handle it differently. I can adjust it but not really sure how.